### ISSUE TITLE
refactor proposal: move webhooks out of `api/v1alpha1` and `api/webhook`

### ISSUE THREAD

**Issue by STRRL** [chaos-mesh/chaos-mesh#2682]
The codes about webhooks are located in several places:

- defaulter/validator(works inside of webhook) registered for certain types, in `api/v1alpha1/genericwebhook`
- webhook for each type of Custom Resource struct, in `api/v1alpha1`, like `api/v1alpha1/podchaos_webhook.go`
- functional webhooks, in `api/webhook`;

I think webhook is not a part of the API definition, so I propose that move them out of `/api`, into another package like `/webhooks`, and only left useful const/enums in `/api`

**Comment by YangKeao** [comment]
I agree with you. When I was splitting the `api/v1alpha` package, I really want to move the webhook part out of it.

However, the problem is that the `controller-runtime` requires the `apiType` to implement `Defaulter` and `Validator` (bad design!).


Here is the code we used to register to webhook:

```go
for _, obj := range params.Objs {
	  if !ccfg.ShouldStartWebhook(obj.Name) {
	  	  continue
	  }
	  
	  err = ctrl.NewWebhookManagedBy(mgr).
		    For(obj.Object).
		    Complete()
	  if err != nil {
	  	  return err
	  }
}
```

And this is how it's implemented inside the controller-runtime:

```go
func (blder *WebhookBuilder) registerValidatingWebhook() {
	validator, isValidator := blder.apiType.(admission.Validator)
	if !isValidator {
		log.Info("skip registering a validating webhook, admission.Validator interface is not implemented", "GVK", blder.gvk)
		return
	}
...
}
```

Though, we could register the webhook manually (like how we register the "/validate-auth"). I still doubt the benefit and trouble it brings.
